contracts/mmr/MMRVerification.sol

```diff
@@ -197,13 +200,79 @@ library MMRVerification {
         require(count == peakIndexes.length, "Invalid bit calculation");
     }
 
-    function numOfPeaks(uint256 width) public pure returns (uint256 num) {
-        uint256 bits = width;
+    /**
+     * @dev Return number of peaks from number of leaves
+     */
+    function numOfPeaks(uint256 numLeaves) public pure returns (uint256 numPeaks) {
+        uint256 bits = numLeaves;
         while (bits > 0) {
-            if (bits % 2 == 1) num++;
+            if (bits % 2 == 1) numPeaks++;
             bits = bits >> 1;
         }
-        return num;
+        return numPeaks;
+    }
+
+    /**
+     * @dev Return MMR size from number of leaves
+     */
+    function getSize(uint256 numLeaves) internal pure returns (uint256) {
+        return (numLeaves << 1) - numOfPeaks(numLeaves);
+    }
+
+    /**
+     * @dev Counts the number of 1s in the binary representation of an integer
+     */
+    function bitCount(uint256 n) internal pure returns(uint256)
+    {
+        uint256 count;
+        while(n > 0) {
+            count = count + 1;
+            n = n & (n-1);
+        }
```

Comment:
Is that the most gas efficient algorithm to achieve this? Iterating over the bits may be faster. No need to address in this PR, but please create a ticket to do some gas benchmarks.

Reply:
Created issue https://github.com/Snowfork/verification-solidity/issues/13 to track this and other gas optimizations to be addressed.